src/daria/analysis/segmentationcomparison.py

```diff
@@ -1,113 +1,471 @@
-            # Overlap of different components
-            return_image[
-                np.logical_or(
-                    np.logical_and(
-                        segmentations[k] == components[0],
-                        segmentations[i] == components[1],
-                    ),
-                    np.logical_and(
-                        segmentations[k] == components[1],
-                        segmentations[i] == components[0],
-                    ),
-                )
-            ] += gray_base[2]
-
-    # Determine locations (and make modifications to return image) of unique components
-    for c_num, c in enumerate(components):
-        for k in range(number_of_segmented_images):
-            only_tmp: np.ndarray = segmentations[k] == c
-            for j in filter(lambda j: j != k, range(number_of_segmented_images)):
-                only_tmp = np.logical_and(
-                    only_tmp, segmentations[j] == non_active_component
+            for i in range(self.number_of_segmented_images):
+                rgba = 255 * np.array(
+                    colormap(1 / self.number_of_segmented_images * (i + 0.5))[0:3]
                 )
+                rgbalight = np.trunc(self.light_scaling * rgba)
+                self.colors[i, 0] = rgba.astype(np.uint8)
+                self.colors[i, 1] = rgbalight.astype(np.uint8)
+        else:
+            colors_pre = kwargs.pop("colors")
+            colors_light: np.ndarray = np.trunc(self.light_scaling * colors_pre)
+            np.clip(colors_light, 0, 255, out=colors_light)
+            self.colors: np.ndarray = np.hstack((colors_pre, colors_light))
+
+        # Assert that there are a sufficient amount of colors
+        # and that all of the segmentations are of equal size
+        assert self.colors.shape[0] == self.number_of_segmented_images
+
+        # Create dictionary with colors and associated situations. Used for legends.
+        self.color_dictionary: dict = {}
+        for i in range(self.number_of_segmented_images):
+            self.color_dictionary[
+                f"Unique apperance of {self.component_names[0]}"
+                f" in {self.segmentation_names[i]}"
+            ] = self.colors[i, 0]
+            self.color_dictionary[
+                f"Unique apperance of {self.component_names[1]}"
+                f" in {self.segmentation_names[i]}"
+            ] = self.colors[i, 1]
+
+        for i in range(self.number_of_segmented_images - 1):
+            self.color_dictionary[
+                f"{i+2} segmentations overlaps in {self.component_names[0]}"
+            ] = self.gray_base[0] * (i + 1)
+            self.color_dictionary[
+                f"{i+2} segmentations overlaps in {self.component_names[1]}"
+            ] = self.gray_base[1] * (i + 1)
+            self.color_dictionary[
+                f"Mixture {self.component_names[0]} and "
+                f"{self.component_names[1]} for {i+2} segmentations"
+            ] = self.gray_base[2] * (i + 1)
+
```

Comment:
This only allows for two consecutive images to overlap. When considering however three or more images, there is also other combinations than 0 and 1, as well as 1 and 2. A simple nested for loop structure should solve the problem.

Reply:
I don't understand what you mean? This is just the setup of the colors, it doesn't actually check which images overlap. Nevertheless, if we really want to distinguish the different gray colors for more than two images I do not think that the code can be kept like this, and we probably need to just choose specific color values for the cases we want to highlight.

Reply:
This was indeed a mess, and I fixed it now.

Reply:
ah OK, I misunderstood the code I think. Sorry. I guess a comment could have helped, but it is fine.